cdap-common/src/main/java/co/cask/cdap/common/kerberos/SecurityUtil.java

```diff
@@ -198,4 +198,10 @@ public static KerberosName getKerberosName(KerberosPrincipalId principalId) {
   public static void validateKerberosPrincipal(KerberosPrincipalId principalId) {
     getKerberosName(principalId);
   }
+
+  public static String getKeytabURIforPrincipal(String principal, CConfiguration cConf) throws IOException {
+    String confPath = cConf.get(Constants.Security.KEYTAB_PATH);
```

Comment:
tricky: it may be that the cConf contains a property with key "name". In that case, the cConf.get would substitute that already... better call getRaw().